Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,251 @@ namespace xr
                     float rawPose[7]{};
                     ArHitResult_getHitPose(xrContext->Session, hitResult, tempPose);
                     ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
-                    HitResult hitResult{};
-                    RawToPose(rawPose, hitResult.Pose);
+                    HitResult hitTestResult{};
+                    RawToPose(rawPose, hitTestResult.Pose);
 
-                    hitResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
-                    filteredResults.push_back(hitResult);
+                    hitTestResult.NativeTrackable = reinterpret_cast<NativeTrackablePtr>(trackable);
+                    filteredResults.push_back(hitTestResult);
                     frameTrackables.push_back(trackable);
                 }
             }
         }
 
+        // Converts an image bitmap to grayscale.
+        // Supported image formats: RBG(A)8, RGB(A)16, 16-bit grayscale.
+        void ConvertBitmapToGrayscale(
+            const uint8_t* imagePixelBuffer,
+            const int32_t width,
+            const int32_t height,
+            const int32_t stride,
+            uint8_t* grayscaleBuffer)
+        {
+            const uint32_t pixelStride = stride / width;
+            for (int h = 0; h < height; ++h)
+            {
+                for (int w = 0; w < width; ++w)
+                {
+                    const uint8_t* pixel{&imagePixelBuffer[w * pixelStride + h * stride]};
```

Comment:
It might be a bit safer and cleaner to use gsl::span here, but up to you.

Reply:
Switched to spans